core/iwasm/libraries/wasi-nn/test/test_tensorflow.c

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <math.h>
+#include <assert.h>
+#include "wasi_nn.h"
+
+#include <fcntl.h>
+#include <errno.h>
+
+#define BUFFER_SIZE 32
+
+#define INPUT_SIZE 25
+
+#define EPSILON 1e-8
+#define MAX_OUTPUT_TENSOR_SIZE 200
+#define MAX_MODEL_SIZE 85000000
+#define MAX_OUTPUT_TENSORS 4
+
+#define GRAPH_EXECUTION_CONTEXT 44
+
+typedef struct {
+    float *input_tensor;
+    uint32_t *dim;
+    uint32_t elements;
+} input_info;
+
+// WASI-NN wrappers
+
+error
+wasm_load(char *model_name, graph *graph)
+{
+    FILE *pFile = fopen(model_name, "r");
+
+    assert(pFile != NULL);
+
+    uint8_t *buffer;
+    size_t result;
+
+    // allocate memory to contain the whole file:
+    buffer = (uint8_t *)malloc(sizeof(uint8_t) * MAX_MODEL_SIZE);
+    if (buffer == NULL)
+        return missing_memory;
+
+    result = fread(buffer, 1, MAX_MODEL_SIZE, pFile);
+    if (result <= 0)
+        return invalid_argument;
+
+    graph_builder_array arr;
+
+    arr.size = 1;
+    arr.buf = (graph_builder *)malloc(sizeof(graph_builder));
+    if (arr.buf == NULL) {
+        free(buffer);
+        return missing_memory;
+    }
+
+    arr.buf[0].size = result;
+    arr.buf[0].buf = buffer;
+
+    error res = load(&arr, tensorflow, cpu, graph);
+
+    fclose(pFile);
+    free(buffer);
+    return res;
```

Comment:
When is `arr.buf` freed? At least it should be freed when load fails. And success, will it be stored in graph or tensorflow?